### PR TITLE
Fix SAML metadata signing

### PR DESCRIPTION
**Pull request by waterkip**
This requires an update on XML::Sig
(see https://github.com/perl-net-saml2/perl-XML-Sig/pull/40)

Fixes: #61

Signed-off-by: Wesley Schwengle <wesley@opndev.io>

### CODE REVIEW

File: lib/Net/SAML2/SP.pm [hunk 1]
```diff
@@ -343,7 +345,12 @@ sub metadata {
                             $ds,
                             $self->_cert_text,
                         )
-                    )
+                    ),
+                    $x->KeyName(
+                     $ds,
+                     Digest::MD5::md5_hex($self->_cert_text)
```
Comment (timlegge) [inline lib/Net/SAML2/SP.pm:351]: Why md5 hex?

**Comment by timlegge** [comment]
Can you rebase?  I see you closed out the XML::Sig PR.  I may pull it together at some point as it was a good feature.

File: lib/Net/SAML2/SP.pm (hunk 1 above)
Comment (waterkip) [reply lib/Net/SAML2/SP.pm:357]: Because it generates a stable "name" for any cert without it being too long or what not.

**Comment by waterkip** [comment]
> Can you rebase? I see you closed out the XML::Sig PR. I may pull it together at some point as it was a good feature.

I accidentally removed the branch while doing some cleaning. I resubmitted it.

**Review by timlegge — APPROVED** [review]
looks fine

**Pull request merged**